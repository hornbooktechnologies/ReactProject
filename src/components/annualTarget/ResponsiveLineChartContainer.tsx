import React from 'react'
import {
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  ResponsiveContainer,
  ReferenceLine,
} from "recharts";
import { ChartConstant ,DateformatConstant } from "../../utils/AppConstants";
import CustomTooltip from "./CustomTooltip";
import CustomLabel from "./CustomLabel";
import { ResponsiveContainerModel } from "../../models/LinechartModel";
import Moment from "moment";

const ResponsiveLineContainer = ({
  data,
  selectValue,
  longestLabelLength,
  ylabel,
  linechartData,
  ForecastList,
  PlannedList,
  ActualList,
  Today,
  list
}:ResponsiveContainerModel) => {
  function formatXAxis(Period) {
    //  Moment.locale('ja');
    let a = Moment(Period);
    switch (selectValue) {
      case ChartConstant.ANNUAL_DATA:
        return a.format(DateformatConstant.MMM);

      case ChartConstant.MONTHLY_DATA:
        return a.format(DateformatConstant.DDD_DO);

      case ChartConstant.WEEKLY_DATA:
        return a.format(DateformatConstant.DDD_DO);

      case ChartConstant.DAILY_DATA:
        return a.format(DateformatConstant.HH_MM);

      case ChartConstant.CUMULATIVE_DATA:
        return a.format(DateformatConstant.MMM);

      default:
        return a.format(DateformatConstant.MMM);
    }
  }
  let date = new Date()
  var presentDate = new Date(date.getTime() - (date.getTimezoneOffset() * 60000 ))
                    .toISOString().split(".")[0]

  return (
    <>
      <ResponsiveContainer width="100%" height="75%">
        <LineChart
          height={273}
          data={data?.map((d) => ({
            ...d,
            
            Planned: d.Planned,
            Forecast: d.Forecast,
            Actual: d.Actual,
           
          }))}
          margin={{
            top: 5,
            right: 30,
            left: 20,
            bottom: 5,
          }}
        >
          <CartesianGrid vertical={false} stroke={ChartConstant.TICKLINE_STROKE} strokeDasharray="3" strokeWidth="0.5"/>
          {selectValue === ChartConstant.MONTHLY_DATA ||
          selectValue === ChartConstant.DAILY_DATA ? (
            <XAxis
              type="category"
              dataKey="Period"
              interval={0}
              tickFormatter={formatXAxis}
              tickSize={10}
              ticks={list}
              tickLine={{
                stroke: ChartConstant.TICKLINE_STROKE,
                width: "1px",
                strokeWidth: 1,
              }}
              style={{ fontSize: "12px"}}
              tick={{ fill: ChartConstant.WHITE_STROKE }}
            />
          ) : (
            <XAxis
              type="category"
              dataKey="Period"
              tickFormatter={formatXAxis}
              tickSize={10}
              tickLine={{
                stroke: ChartConstant.TICKLINE_STROKE,
                width: "1px",
                strokeWidth: 1,
              }}
              tick={{ fill: ChartConstant.WHITE_STROKE }}
              style={{ fontSize: "12px" }}
            />
          )}
          {selectValue === ChartConstant.MONTHLY_DATA ||
          selectValue === ChartConstant.DAILY_DATA ? (
            <XAxis
              type="category"
              dataKey="Period"
              interval={0}
              xAxisId="0"
              tickFormatter={(x) => ""}
              tickLine={{
                stroke: ChartConstant.TICKLINE_STROKE,
                width: "1px",
                strokeWidth: 1,
              }}
              tickSize={10}
              tick={{ fill: ChartConstant.WHITE_STROKE }}
              style={{ fontSize: "12px" }}
            />
            ) : null}

          <XAxis
            type="number"
            dataKey="TimeStamp"
            hide={true}
            domain={["dataMin", "dataMax"]}
            xAxisId="ref"
            tickFormatter={formatXAxis}
          />

          <YAxis
            type="number"
            axisLine={false}
            tick={{ fill: ChartConstant.WHITE_STROKE }}
            width={longestLabelLength * 10}
            label={{
              value: ylabel,
              fill: ChartConstant.WHITE_STROKE,
              angle: -90,
              position: "insideLeft",
            }}
            style={{ fontSize: "12px" }}
          ></YAxis>
          <Tooltip content={<CustomTooltip selectValue={selectValue} linechartData={linechartData} active={false} payload={[]}/>} />

          {ForecastList?.length > 0 ? (
            <Line
              dataKey="Forecast"
              xAxisId="0"
              stroke={ChartConstant.FORECAST_LINE_STROKE}
              strokeWidth="3"
              dot={{
                r: 5,
                stroke: ChartConstant.WHITE_STROKE,
                fill: ChartConstant.FORECAST_COLOR,
                strokeWidth: 1,
              }}
              activeDot={{ r: 5 }}
            />
          ) : null}
          {PlannedList?.length > 0 ? (
            <Line
              dataKey="Planned"
              xAxisId="0"
              stroke={ChartConstant.PLANNED_LINE_STROKE}
              strokeWidth="3"
              dot={{
                r: 5,
                stroke: ChartConstant.WHITE_STROKE,
                fill: ChartConstant.PLANNED_COLOR,
                strokeWidth: 1,
              }}
              activeDot={{ r: 5 }}
            />
          ) : null}
          {ActualList?.length > 0 ? (
            <Line
              dataKey="Actual"
              xAxisId="0"
              stroke={ChartConstant.ACTUAL_LINE_STROKE}
              strokeWidth="3"
              dot={{
                r: 5,
                stroke: ChartConstant.WHITE_STROKE,
                fill: ChartConstant.ACTUAL_COLOR,
                strokeWidth: 1,
              }}
              activeDot={{ r: 5 }}
            />
          ) : null}

          <ReferenceLine
            xAxisId="ref"
            x={new Date(presentDate).getTime()}
            stroke={ChartConstant.LABEL_COLOR}
            strokeWidth={1.5}
            label={<CustomLabel viewBox={{x: 0, y: 0}}/>}
          />
        </LineChart>
      </ResponsiveContainer>
    </>
  );
};
export default ResponsiveLineContainer;
