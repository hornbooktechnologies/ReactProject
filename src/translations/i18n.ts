import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
