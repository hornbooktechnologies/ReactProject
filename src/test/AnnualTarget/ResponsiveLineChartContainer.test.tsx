import React from "react";
import ResponsiveLineChartContainer from "../../components/annualTarget/ResponsiveLineChartContainer";
import renderer from "react-test-renderer";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";
describe("Component: ResponsiveLineChartContainer", () => {
  const props = {
    linechartData: {
      Prefix: "¥",
      Suffix: "Oku",
      Suffix2: "Oku",
      currentDisable: false,
      datelabel: "2021",
      graphData:
        '{"Actual":[3.07,1.99,6.14,1.09,null,null,null,null,null,null,null,null],"Period":["2021-04-01T00:00","2021-05-01T00:00","2021-06-01T00:00","2021-07-01T00:00","2021-08-01T00:00","2021-09-01T00:00","2021-10-01T00:00","2021-11-01T00:00","2021-12-01T00:00","2022-01-01T00:00","2022-02-01T00:00","2022-03-01T00:00"],"Planned":[3.99,1.86,2.76,2.5,23.46,9.83,1.94,1.88,2.88,1.79,2.11,3.49],"Forecast":[null,null,null,null,1.43,2.64,1.67,1.59,2.42,1.64,1.71,3.33],"Estimates":[58.54,12.3,16.47]}',
      indexFound: true,
      nextDisable: false,
    },
    list: [],
    longestLabelLength: 6,
    selectValue: "AnnualData",
    ylabel: "¥ Oku",
    ActualList: [3.07, 1.99, 6.14, 1.09],
    ForecastList: [1.43, 2.64, 1.67, 1.59, 2.42, 1.64, 1.71, 3.33],
    PlannedList: [
      3.99,
      1.86,
      2.76,
      2.5,
      23.46,
      9.83,
      1.94,
      1.88,
      2.88,
      1.79,
      2.11,
      3.49,
    ],
    Today: "2022-03-01T00:00",
    data: [
      {
        Actual: 3.07,
        Forecast: null,
        Period: "2021-04-01T00:00",
        Planned: 3.99,
        TimeStamp: 1617215400000,
      },
      {
        Actual: 3.07,
        Forecast: null,
        Period: "2021-05-01T00:00",
        Planned: 3.99,
        TimeStamp: 1617215400000,
      },
      {
        Actual: 3.07,
        Forecast: null,
        Period: "2021-06-01T00:00",
        Planned: 3.99,
        TimeStamp: 1617215400000,
      },
    ],
  };
  it("ResponsiveLineChartContainer renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <ResponsiveLineChartContainer {...props}/>
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
