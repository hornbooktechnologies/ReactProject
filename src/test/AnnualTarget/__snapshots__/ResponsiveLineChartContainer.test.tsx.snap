// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ResponsiveLineChartContainer ResponsiveLineChartContainer renders correctly 1`] = `
<div
  className="recharts-responsive-container"
  style={
    Object {
      "height": "75%",
      "maxHeight": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "width": "100%",
    }
  }
/>
`;
